src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcConfirmationTokenDao.java

```diff
@@ -0,0 +1,122 @@
+package com.vinylteam.vinyl.dao.jdbc;
+
+import com.vinylteam.vinyl.dao.ConfirmationTokenDao;
+import com.vinylteam.vinyl.dao.RowMapper;
+import com.vinylteam.vinyl.dao.jdbc.mapper.ConfirmationTokenRowMapper;
+import com.vinylteam.vinyl.entity.ConfirmationToken;
+import com.zaxxer.hikari.HikariDataSource;
+import lombok.extern.slf4j.Slf4j;
+
+import java.sql.*;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+
+@Slf4j
+public class JdbcConfirmationTokenDao implements ConfirmationTokenDao {
+
+    private static final String INSERT_WHEN_USER_STATUS = "INSERT INTO confirmation_tokens (user_id, token, created_at) SELECT ?, ?, ? " +
+            "WHERE EXISTS (SELECT * FROM users WHERE id=? AND status=FALSE)";
+    private static final String SELECT_BY_TOKEN = "SELECT id, user_id, token, created_at FROM confirmation_tokens WHERE token=?";
+    private static final String SELECT_BY_USER_ID = "SELECT id, user_id, token, created_at FROM confirmation_tokens WHERE user_id=?";
+    private static final String DELETE_BY_USER_ID_WHEN_USER_STATUS = "DELETE FROM confirmation_tokens WHERE user_id=? AND " +
+            "EXISTS (SELECT * FROM users WHERE users.id=? AND users.status)";
+    private static final RowMapper<ConfirmationToken> ROW_MAPPER = new ConfirmationTokenRowMapper();
+    private final HikariDataSource dataSource;
+
+    public JdbcConfirmationTokenDao(HikariDataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public Optional<ConfirmationToken> findByToken(UUID token) {
+        ConfirmationToken confirmationToken = null;
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement findByEmailStatement = connection.prepareStatement(SELECT_BY_TOKEN)) {
+            findByEmailStatement.setObject(1, token);
+            log.debug("Prepared statement {'preparedStatement':{}}.", findByEmailStatement);
+            try (ResultSet resultSet = findByEmailStatement.executeQuery()) {
+                if (resultSet.next()) {
+                    confirmationToken = ROW_MAPPER.mapRow(resultSet);
+                }
+            }
+        } catch (SQLException e) {
+            log.error("SQLException retrieving confirmation token by email from confirmation_tokens", e);
+            throw new RuntimeException(e);
+        }
+        log.debug("Resulting optional with confirmation token is {'Optional.ofNullable(confirmationToken)':{}}", Optional.ofNullable(confirmationToken));
+        return Optional.ofNullable(confirmationToken);
+    }
```

Comment:
Which two rows are you referring to?
We can't place confirmationToken = null inside try: if try block errors, confirmationToken wouldn't be initialised, because everything that is initialised within try {} would be gone.

Reply:
47, 48 I would put inside try